sklearn/utils/fixes.py

```diff
@@ -220,3 +220,51 @@ def __init__(self, function):
     def __call__(self, *args, **kwargs):
         with config_context(**self.config):
             return self.function(*args, **kwargs)
+
+
+def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
```

Comment:
is it just a copy paste from numpy code?

Reply:
`numpy.linspace` implementation uses a `array_function_dispatch` decorator. Just copy & pasting that would have introduced a lot of (boilerplate) code. But yes, about half of the code is copied from numpy.